File: backend-spring/src/main/java/io/github/agussyahrilmubarok/backend/service/IUserService.java
package io.github.agussyahrilmubarok.backend.service;

import io.github.agussyahrilmubarok.backend.model.user.CreateUserRequest;
import io.github.agussyahrilmubarok.backend.model.user.UpdateUserRequest;
import io.github.agussyahrilmubarok.backend.model.user.UserResponse;

import java.util.List;

public interface IUserService {
    List<UserResponse> findAll();

    UserResponse findById(String userId);

    UserResponse create(CreateUserRequest request);

    UserResponse updateById(String userId, UpdateUserRequest request);

    void deleteById(String userId);
}

File: backend-dotnet/Backend/Application/Service/IUserService.cs
using Backend.Application.DTO.User;

namespace Backend.Application.Service;

public interface IUserService
{
    Task<IEnumerable<UserResponse>> GetAllAsync(CancellationToken ct = default);
    Task<UserResponse> GetByIdAsync(Guid id, CancellationToken ct = default);
    Task<UserResponse> CreateAsync(CreateUserRequest request, CancellationToken ct = default);
    Task<UserResponse> UpdateAsync(Guid id, UpdateUserRequest request, CancellationToken ct = default);
    Task DeleteAsync(Guid id, CancellationToken ct = default);
}